src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -293,6 +304,10 @@ public override bool RunTask ()
 
 			existingEntries.Clear ();
 
+			foreach (var pattern in ExcludeFiles ?? Array.Empty<string> ()) {
+				excludePatterns.Add (new Regex (pattern.Replace ("*", "(.)"), RegexOptions.IgnoreCase | RegexOptions.Compiled));
```

Comment:
I don't understand this `.Replace()`?  Why replace `*` with `(.)`?

Reply:
Realizing that `@(AndroidPackagingOptionsExclude)` contains *file globs*, it makes sense that `*` needs to be replaced, but replacing with `(.)` still doesn't make sense to me.  Wouldn't `.*` make more sense?

On a similar angle, the documentation for "search pattern" (see above `Directory.EnumerateFiles()` doc link) says:

> parameter can contain a combination of valid literal path and wildcard (`*` and `?`) characters, but it doesn't support regular expressions.

This "file glob to Regex" conversion doesn't handle `?`.

We should probably have something like:

```csharp
static Regex GlobToRegex (string glob)
{
    var r = new StringBuilder (glob.Length);
    foreach (var c in glob) {
        switch (c) {
        case '*': r.Append (".*");  break;
        case '?': r.Append (".");   break;
        case '.': r.Append (@"\."); break;
        default:  r.Append (c);
        }
    }
    return new Regex (r.ToString ());
}
```